File: petgo_app/lib/shared/boundary/triage_category_jump.dart
/// 🔴🔴 问诊与商品之间**唯一被允许的关联**（Story 9.1 · FR-110 · **安全攸关**）。
///
/// FR-110 只开了一个口子：
/// > 仅允许「按结论中的结构化健康记录类型（如驱虫）跳转对应品类」这一种形式，**且由系统生成**。
/// > 🔴 兽医无法选择具体 SKU，无任何界面入口。
///
/// 所以这里的输入是**健康记录类型 code**、输出是**品类 code**，两头都是受控枚举串。
/// 🔴 **整条路径上不存在任何能携带商品/SKU 的形参** —— 这就是「能力缺席」：
/// 不是「兽医没权限选 SKU」，而是**根本没有一个地方能把 SKU 传进来**。
/// 权限能改、能绕；一个不存在的参数改不了也绕不过。
///
/// 后端同名守卫见 `com.tailtopia.shared.boundary.TriageCategoryJump`。
///
/// ⚠️ **刻意用 String 而不是 `ShopCategory`**：一旦这里 import `features/shop`，
/// 调用方就有了一条通往商品域的路。桥只让一个 code 通过，比让一个类型通过更难被撑大。
library;

class TriageCategoryJump {
  const TriageCategoryJump._();

  /// 品类 code（与后端 `ProductCategory` / `ShopCategory.api` 同一套字面量）。
  static const String _obatVitamin = 'OBAT_VITAMIN';

  /// 健康记录类型 code → 品类 code。**不在表里的一律返回 null（不跳）**。
  ///
  /// 🔴 只有驱虫与疫苗有对应品类。其余类型（生理期 / 绝育 / 自定义）
  /// **刻意不给跳转** —— 「每条结论都配一个购物入口」正是 FR-110 要防的形态。
  static String? categoryFor(String? recordType) => switch (recordType) {
        'DEWORM' || 'VACCINE' => _obatVitamin,
        _ => null,
      };

  static bool allowsJump(String? recordType) => categoryFor(recordType) != null;
}

File: petgo-backend/src/main/java/com/tailtopia/shared/boundary/TriageCategoryJump.java
package com.tailtopia.shared.boundary;

import com.tailtopia.profile.domain.HealthRecordType;
import com.tailtopia.shop.domain.ProductCategory;

/**
 * 🔴 <b>问诊结论页唯一允许的商品关联</b>（Story 9.1，FR-110）。
 *
 * <p>规则一句话：<b>按结论中的结构化健康记录类型跳转对应品类，且由系统生成。</b>
 *
 * <p>🔴 <b>兽医无法选择具体 SKU —— 不是"有入口但被权限拦住"，是根本没有那个入口</b>：
 * 本类的输入是 {@link HealthRecordType}（一个枚举），输出是 {@link ProductCategory}（另一个枚举）。
 * <b>整条路径上不存在任何可以携带 SKU 的地方</b>，这就是"能力缺席"的实现形态。
 *
 * <p>⚠️ 埋点 {@code triage_to_category_jump}（含 {@code record_type}）用于监控<b>边界侵蚀</b>：
 * 问诊 → 商品的跳转占比若持续走高，说明问诊正在被当作销售前端使用 —— 那是要立刻叫停的信号，
 * 不是要庆祝的转化。
 */
public final class TriageCategoryJump {

    private TriageCategoryJump() {
    }

    /**
     * 结构化健康记录类型 → 商品品类。
     *
     * <p>🔴 <b>只映射两类</b>（驱虫 / 疫苗后的护理用药），其余一律返回 {@code null} = <b>不跳</b>。
     * 把「月经 / 绝育 / 自定义」也映射过去，就等于给每一条问诊结论都配一个购物入口 ——
     * 那正是 FR-110 要防的形态。
     */
    public static ProductCategory categoryFor(HealthRecordType recordType) {
        if (recordType == null) {
            return null;
        }
        return switch (recordType) {
            case DEWORM, VACCINE -> ProductCategory.OBAT_VITAMIN;
            // 🔴 其余类型不跳。默认分支写成 null 而不是某个"兜底品类"是刻意的。
            default -> null;
        };
    }

    /** 该结论是否允许出现品类跳转入口。 */
    public static boolean allowsJump(HealthRecordType recordType) {
        return categoryFor(recordType) != null;
    }
}
